packages/menu/src/SubMenu.tsx

```diff
@@ -0,0 +1,349 @@
+import { HTMLElementProps, createDataProp } from '@leafygreen-ui/lib';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import {
+  menuItemContainerStyle,
+  activeMenuItemContainerStyle,
+  disabledMenuItemContainerStyle,
+  focusedMenuItemContainerStyle,
+  linkStyle,
+  disabledTextStyle,
+} from './styles';
+import { ExitHandler } from 'react-transition-group/Transition';
+
+const subMenuContainer = createDataProp('sub-menu-container');
+
+const liStyle = css`
+  position: relative;
+  overflow: hidden;
+`;
+
+const subMenuStyle = css`
+  flex-direction: row;
+  min-height: 56px;
+  border-top: 1px solid ${uiColors.gray.light2};
+  background-color: ${uiColors.gray.light3};
+  align-items: center;
+  justify-content: flex-start;
+`;
+
+const subMenuOpenStyle = css`
+  background-color: ${uiColors.white};
+  &:hover {
+    background-color: ${uiColors.gray.light2};
+  }
+`;
+
+const closedIconStyle = css`
+  transition: color 200ms ease-in-out;
+  color: ${uiColors.gray.base};
+`;
+
+const openIconStyle = css`
+  color: ${uiColors.gray.dark2};
+`;
+
+const mainTextStyle = css`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  font-size: 14px;
+  color: ${uiColors.gray.dark3};
+  ${subMenuContainer.selector}:focus & {
+    color: ${uiColors.blue.dark3};
+  }
+`;
+
+const activeMainTextStyle = css`
+  font-weight: bolder;
+  color: ${uiColors.green.dark3};
+`;
+
+const iconButtonStyle = css`
+  position: absolute;
+  z-index: 1;
+  right: 8px;
+  top: ${52 / 2 - 22 / 2}px;
+  margin: auto;
+  background-color: ${uiColors.gray.light3};
+  transition: background-color 150ms ease-in-out;
+  ${subMenuContainer.selector}:hover + & {
+    background-color: ${uiColors.gray.light2};
+    transition: background-color 150ms ease-in-out;
+  }
+`;
+
+const iconButtonFocusedStyle = css`
+  ${subMenuContainer.selector}:focus + & {
+    background-color: ${uiColors.blue.light3};
+    transition: background-color 150ms ease-in-out;
+  }
+`;
+
+const openIconButtonStyle = css`
+  background-color: ${uiColors.white};
+  transition: background-color 150ms ease-in-out;
+`;
+
+const mainIconStyle = css`
+  color: ${uiColors.gray.base};
+  // 15px of padding from MenuItem + 32px width of SVG + 13px = 60px
+  // which is the padding-left for children of SubMenus.
+  margin-right: 13px;
+  flex-shrink: 0;
+  ${subMenuContainer.selector}:hover > & {
+    color: ${uiColors.gray.dark1};
+  }
+`;
+
+const mainIconFocusedStyle = css`
+  ${subMenuContainer.selector}:focus > & {
+    color: ${uiColors.blue.base};
+  }
+`;
+
+const activeIconStyle = css`
+  color: ${uiColors.green.base};
+  ${subMenuContainer.selector}:hover > & {
+    color: ${uiColors.green.base};
+  }
+`;
+
+const ulStyle = css`
+  list-style: none;
+  padding: 0;
+  height: 0;
+  overflow: hidden;
+  transition: height 150ms ease-in-out;
+`;
+
+const menuItemBorder = css`
+  position: absolute;
+  width: 100%;
```

Comment:
With absolute-positioned elements, we should use:
```css
left: 0;
right: 0;
```

Reply:
left: 0 makes the border start all the way at the left; we want the border to be more indented and not fill the entire line

Reply:
![Screen Shot 2020-01-21 at 3 42 07 PM](https://user-images.githubusercontent.com/26016393/72841562-a15f6080-3c64-11ea-9b86-7deadf17e94a.png)

Reply:
^ hard to articulate

Reply:
Yeah I actually played around with this a lot locally. What is here isn't great but I can't think of anything that's drastically better.

Reply:
~Hmm... So I got it working, though I had to make a few other CSS tweaks to its ancestors.~

Nevermind, that created other problems. Carry on! 😅